tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1499,6 +1499,10 @@ class DxcCompiler : public IDxcCompiler3,
     compiler.getLangOpts().EnablePayloadAccessQualifiers = Opts.EnablePayloadQualifiers;
     compiler.getLangOpts().HLSLProfile =
           compiler.getCodeGenOpts().HLSLProfile = Opts.TargetProfile;
+
+    compiler.getCodeGenOpts().HLSLEnablePartialLifetimeMarkers =
```

Comment:
The way that some of these toggles are used, particularly with CodeGenOpts, is ugly and inconsistent.  I think these could be improved, but not necessarily in this PR.  Here are some thoughts:

For access of toggles: It seems we could abstract the toggles in a class that contains the toggle map, and access using a `Get()` with a default value parameter.  This `Get()` wouldn't have to look up the value twice.  Additionally, we could define the string names of front-end referenced toggles in one place, and refer to these using an enum defined in the toggle class's namespace.  `Get()` could then have an overload for the enum.

For `CodeGenOpts`, we want to refer to boolean values to initialize the pass manager, or for any other front-end purpose, I think we should have dedicated boolean option values for these states, which would be initialized by any relevant toggles early, outside the context of pass manager initialization (`BackendUtil.cpp`).

Finally, I think we should refactor the various component option initializations into functions that this `SetupCompilerForCompile` function uses, such as: `InitLangOptsForHLSL()`, `InitCodeGenOptsForHLSL()`, and `InitPPOptsForHLSL()`.

Reply:
Yeah agreed. It should also be easier to read at a glance if an opt toggle is default true of default false.